pkgs/applications/blockchains/bisq-desktop/default.nix

```diff
@@ -0,0 +1,104 @@
+{ stdenv, lib, callPackage, makeWrapper, fetchurl, rsync 
+, protobuf3_10
+, makeDesktopItem
+, imagemagick
+}:
+let
+  bisq-launcher = callPackage ./launcher.nix {};
+  common = callPackage ./common.nix {};
+
+  # This takes all of the Java dependencies from deps.nix,
+  # and creates a single mvn directory tree so that gradle
+  # can find the project's dependencies.
+  deps = let
+    mkDepDerivation = { name, url, sha256, mavenDir }: 
+      stdenv.mkDerivation {
+        name = "bisq-dep-${name}";
+
+        src = fetchurl { inherit sha256 url; };
+
+        unpackCmd = ''
+          mkdir output 
+          cp $curSrc "output/${name}"
+        '';
+
+        installPhase = ''
+          mkdir -p "$out/${mavenDir}"
+          cp -r . "$out/${mavenDir}/"
+        '';
+      };
+
+    d = map mkDepDerivation (import ./deps.nix); 
+    in stdenv.mkDerivation {
+      name = "bisq-deps";
+      dontUnpack = true;
+
+      installPhase = ''
+        mkdir $out
+
+        for p in ${toString d}; do
+          ${rsync}/bin/rsync -a $p/ $out/
```

Comment:
Why are we not using cp here?

Reply:
I tried `cp` but it didn't work.

Reply:
To be more specific, I could not get `cp` to create a proper Maven directory tree.

Reply:
I need more details if I should help you but I am pretty sure cp will work here but you maybe need to try with ``./`` or ``/`` a bit.